contracts/gauge-borrowing/VaultPriceOracle.sol

```diff
@@ -0,0 +1,188 @@
+     * @dev Changes the admin and emits an event.
+     */
+    function changeAdmin(address newAdmin) external onlyAdmin {
+        address oldAdmin = admin;
+        admin = newAdmin;
+        emit NewAdmin(oldAdmin, newAdmin);
+    }
+
+    /**
+     * @dev Event emitted when `admin` is changed.
+     */
+    event NewAdmin(address oldAdmin, address newAdmin);
+
+    /**
+     * @dev Modifier that checks if `msg.sender == admin`.
+     */
+    modifier onlyAdmin {
+        require(msg.sender == admin, "Sender is not the admin.");
+        _;
+    }
+
+    /**
+     * @dev Admin-only function to set price feeds.
+     * @param underlyings Underlying token addresses for which to set price feeds. (gauge token)
+     * @param feeds The Chainlink price feed contract addresses for each of `underlyings`. (underlying of gauge's vault token)
+     * @param baseCurrency The currency in which `feeds` are based.
+     */
+    function setPriceFeeds(address[] memory underlyings, IAggregatorV3Interface[] memory feeds, FeedBaseCurrency baseCurrency) external onlyAdmin {
+        // Input validation
+        require(underlyings.length > 0 && underlyings.length == feeds.length, "Lengths of both arrays must be equal and greater than 0.");
+
+        // For each token/feed
+        for (uint256 i = 0; i < underlyings.length; i++) {
+            address underlying = underlyings[i];
+
+            // Check for existing oracle if !canAdminOverwrite
+            if (!canAdminOverwrite) require(address(priceFeeds[underlying]) == address(0), "Admin cannot overwrite existing assignments of price feeds to underlying tokens.");
+
+            // Set feed and base currency
+            priceFeeds[underlying] = feeds[i];
+            feedBaseCurrencies[underlying] = baseCurrency;
+        }
+    }
+
+    /**
+     * @dev Internal function returning the price in ETH of `underlying`.
+     */
+    function _price(address underlying) internal view returns (uint) {
+        // Get token/ETH price from Chainlink
+        IAggregatorV3Interface feed = priceFeeds[underlying];
+        require(address(feed) != address(0), "No Chainlink price feed found for this underlying ERC20 token.");
+        FeedBaseCurrency baseCurrency = feedBaseCurrencies[underlying];
+
+        IRibbonVault vault = IRibbonVault(ILiquidityGauge(underlying).lp_token());
+        console.log("address(vault) is %s", address(vault));
+        uint256 rVaultDecimals = vault.decimals();
+        uint256 rVaultToAssetExchangeRate = vault.pricePerShare(); // (ex: rETH-THETA -> ETH, rBTC-THETA -> BTC)
+
+        // underlying = rETH-THETA-gauge
+        // vault = rETH-THETA
+        // feed = ETH (underlying asset of vault rETH-THETA)
+        // underlying price = feed * (vault token to asset of vault exchange rate)
+
+        // rETH-THETA-gauge -> rETH-THETA -> ETH
+
+        if (baseCurrency == FeedBaseCurrency.ETH) {
+            // If ETH or stETH vault gauge
+            if (address(feed) == address(1)) {
```

Comment:
Why would feed be assigned to `address(1)`?

Reply:
Maybe could be a function called `isEthOrStEThVault()`?

Reply:
Also stETH trades at a premium, not sure if we accounted for that or intentionally ignoring it.

Reply:
it is accounted for in the price per share

Reply:
address(1) signifies it is an eth vault. you think we need _isEthOrStEThVault(feed) {feed == address(1)} ?

Reply:
What about a constant to avoid magic numbers?  
```
address ETH_VAULT = address(1);
```